### PR TITLE
TDD

### PR DESCRIPTION
**Pull request by hinthornw**
Also consolidate some how-tos + condense titles.

Obviously not enough but slight step in the right direction i think.

The problem with this one is it mixes "tutorial", "explanation", and "how-to" into one doc and put in the how-tos section... Not great organization

### CODE REVIEW

File: docs/evaluation/faq/evaluator-implementations.mdx [hunk 1]
```diff
@@ -1,10 +1,11 @@
 ---
-sidebar_label: Use off-the-shelf LangChain evaluators 
-sidebar_position: 6
+sidebar_label: Use Open Source Evaluators
```
Comment (samnoyes) [inline docs/evaluation/faq/evaluator-implementations.mdx:2]: synced offline: change back to off-the-shelf

File: docs/evaluation/faq/manage-datasets.mdx [hunk 2]
```diff
@@ -10,7 +10,50 @@ import {
...
+
+## In LangSmith
+
+The easiest way to interact with datasets is directly in the LangSmith app. Here, you can create and edit datasets and example rows. Below are a few ways to interact with them.
+
+### From Existing Runs
+
+We typically construct datasets over time by collecting representative examples from debugging or other runs. To do this, we first filter the runs to find the ones we want to add to the dataset. Then, we create a dataset and add the runs as examples.
+
+You can do this from any 'run' details page by clicking the 'Add to Dataset' button in the top right-hand corner.
```
Comment (samnoyes) [inline docs/evaluation/faq/manage-datasets.mdx:23]: Maybe we could add a screenshot of the "bulk add" here as well?

Comment (hinthornw) [reply docs/evaluation/faq/manage-datasets.mdx:23]: ya - i'm gonna wait for another pr for that since i do light mode and feel like they should be consistent

File: docs/evaluation/faq/custom-evaluators.mdx [hunk 3]
```diff
@@ -154,7 +157,39 @@ def faithfulness(run: Run, example: Example) -> dict:
     return {"key": "faithfulness", "score": average_score, "comment": comment}
 ```
 
-#### Example 4: Perplexity Evaluator
+#### Example 4: Returning Multiple Scores
+
+A single evaluator can return multiple scores. An example of when this might be useful is if you are using tool calling for an LLM-as-judge to extract multiple metrics in a single API call.
+
+```python
+from langsmith.schemas import Example, Run
```
Comment (agola11) [inline docs/evaluation/faq/custom-evaluators.mdx:165]: would stray away from langchain examples in the docs

File: docs/evaluation/faq/unit-testing.mdx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+LANGCHAIN_TEST_CACHE=tests/cassettes ptw tests/my_llm_tests  
+```
+
+## Conclusion
+
+Unit testing with LangSmith is a quick and effective way to catch bugs and regressions in your AI system. By instrumenting your test functions with `@unit`, you can automatically track test results as experiments, monitor performance on specific test cases, and catch issues before they make it to production. Tests are fast and cheap to write, making them an ideal starting point for teams that want to begin systematically testing their LLMs without a huge upfront investment. Try adding LangSmith to your test suite today and start gaining visibility into your model's behavior.
+
+:::note
+`@unit` is currently only supported in python version `>=0.1.42`. If you are interested in unit testing functionality in TypeScript or other languages, please let us know at [support@langchain.dev](mailto:support@langchain.dev).  
+:::
```
Comment (agola11) [inline docs/evaluation/faq/unit-testing.mdx:131]: add this to the top

File: docs/evaluation/faq/unit-testing.mdx [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+With caching enabled, you can iterate quickly on your tests using `watch` mode without worrying about unnecessarily hitting your LLM provider. For example, using [`pytest-watch`](https://pypi.org/project/pytest-watch/):
+
+```bash
+pip install pytest-watch
+LANGCHAIN_TEST_CACHE=tests/cassettes ptw tests/my_llm_tests  
+```
+
+## Conclusion
+
+Unit testing with LangSmith is a quick and effective way to catch bugs and regressions in your AI system. By instrumenting your test functions with `@unit`, you can automatically track test results as experiments, monitor performance on specific test cases, and catch issues before they make it to production. Tests are fast and cheap to write, making them an ideal starting point for teams that want to begin systematically testing their LLMs without a huge upfront investment. Try adding LangSmith to your test suite today and start gaining visibility into your model's behavior.
```
Comment (agola11) [inline docs/evaluation/faq/unit-testing.mdx:127]: this part feels like fluff, would cut out

File: docs/evaluation/faq/unit-testing.mdx [hunk 6]
```diff
@@ -0,0 +1,131 @@
+---
+sidebar_label: Unit Test 
+sidebar_position: 3
+---
+
+# Unit Tests
+
+Unit tests for LLMs are assertions and checks designed to **quickly** identify obvious bugs and regressions in your AI system. Relative to evaluations, tests are designed to be **fast** and **cheap** to run, focusing on **specific** functionality and edge cases.
```
Comment (agola11) [inline docs/evaluation/faq/unit-testing.mdx:8]:
```suggestion
LangSmith unit tests for LLM applications are assertions and checks designed to **quickly** identify obvious bugs and regressions in your AI system. Relative to evaluations, tests are designed to be **fast** and **cheap** to run, focusing on **specific** functionality and edge cases. 
```

**Review by agola11 — APPROVED** [review]

File: docs/evaluation/faq/custom-evaluators.mdx (hunk 3 above)
Comment (hinthornw) [reply docs/evaluation/faq/custom-evaluators.mdx:165]: alright, will use the client directly
